pkg/controller/perconaservermongodb/users.go

```diff
@@ -145,74 +148,74 @@ func (su *systemUsers) len() int {
 	return len(su.users)
 }
 
-func (r *ReconcilePerconaServerMongoDB) updateSysUsers(cr *api.PerconaServerMongoDB, newUsersSec, currUsersSec *corev1.Secret) (restartSfs bool, err error) {
+func (r *ReconcilePerconaServerMongoDB) updateSysUsers(cr *api.PerconaServerMongoDB, newUsersSec, currUsersSec *corev1.Secret, repls []*api.ReplsetSpec) (restartSfs, restartMongos bool, err error) {
 	su := systemUsers{
 		currData: currUsersSec.Data,
 		newData:  newUsersSec.Data,
 	}
 
 	type user struct {
-		nameKey, passKey string
-		needRestart      bool
+		nameKey, passKey  string
+		needRestartSfs    bool
+		needRestartMongos bool
 	}
 	users := []user{
 		{
-			nameKey: envMongoDBClusterAdminUser,
-			passKey: envMongoDBClusterAdminPassword,
+			nameKey:           envMongoDBClusterAdminUser,
+			passKey:           envMongoDBClusterAdminPassword,
+			needRestartMongos: true,
```

Comment:
hm, why it is needed? mongos do authentication via certificate or key, not user
do you see any particular error?

Reply:
we pass clusterAdmin user to mongos by parameters when we start it, and its use it to add shards

Reply:
comment added in https://github.com/percona/percona-server-mongodb-operator/pull/462/commits/49183156657033fd5c7e34f585c34f6023ae2475